src/controllers/midi/legacymidicontrollermappingfilehandler.cpp

```diff
@@ -273,8 +278,11 @@ QDomElement LegacyMidiControllerMappingFileHandler::inputMappingToXML(
         QDomDocument* doc, const MidiInputMapping& mapping) const {
     QDomElement controlNode = doc->createElement("control");
 
-    controlNode.appendChild(makeTextElement(doc, "group", mapping.control.group));
-    controlNode.appendChild(makeTextElement(doc, "key", mapping.control.item));
+    assert((void("Can't produce XML from a Javascript registered handler"),
+            holds_alternative<QJSValue>(mapping.control)));
```

Comment:
I think it would make sense to check this at the very start and then either DEBUG_ASSERT (or VERIFY_OR_DEBUG_ASSERT) (see `util/debug.h`) or just plain return without creating an control element (not sure if thats possible with the way this currently implemented).